rcl/src/rcl/rmw_implementation_identifier_check.c

```diff
@@ -63,16 +88,62 @@ INITIALIZER(initialize) {
     )
     exit(ret);
```

Comment:
This leaks `expected_rmw_impl`. It needs to be freed on early exit.

Same below.

Reply:
I considered it but didn't see the point if we're exiting. Could you clarify why it's needed?

Reply:
Because otherwise the previously allocated memory is being leaked and can not be recovered.

Reply:
I understand the concept if this were a recoverable state, but it's not recoverable so won't the memory be automatically freed when the process dies (which we're immediately triggering)?

Reply:
Oh, I see, it is actually terminating the process. Then you actually don't have to bother. I was mistaking `exit` for something custom. Sorry for the noise.